app/components/markdown.tsx

```diff
@@ -109,7 +109,7 @@ function escapeDollarNumber(text: string): string {
     if (!isInBlockCode) {
       return line.split(/(`.*?`)/g).map((segment, index) => {
         if (index % 2 === 0) {
-          return segment.replace(/(?<!\\)\$\d+([,.](\d+[,.])?\d+)?(?!.*\$\B)/g, '\\$&');
+          return segment.replace(/(?<!\\)\$\d+([,.](\d+[,.])?\d+)?(?!.*\$\.)/g, '\\$&');
```

Comment:
better

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/a1af7263-63db-4ece-86b9-740af23654ed)

Reply:
Unfortunately, there is a new issue with: $1$, or any other number. It always escapes numbers. I will look into it.

Reply:
> Unfortunately, there is a new issue with: `$1$`, or any other number. It always escapes numbers. I will look into it.

@Algorithm5838 you right

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/33493c34-820e-4011-b4fe-9b60aaf0ccf8)


here is the pattern for a test

```
$1$ $1$
$1 $2 $3 $3
`$1` $2 $3 `$4`
$x$ $hello world$  $x$  $1 $2 $3 $4
$x$ $hello world$  $x$  $1 $2 $3 $4 $x$ $hello world$  $x$
$1 $2 $3 $4 $x$
```
```
$1 2 $3 $4 $5
```

Reply:
Fixed.